src/PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/Ricci.cpp

```diff
@@ -0,0 +1,163 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/Ricci.hpp"
+
+#include "DataStructures/DataVector.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tags/TempTensor.hpp"
+#include "DataStructures/TempBuffer.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/ContainerHelpers.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+/// \cond
+namespace GeneralizedHarmonic {
+template <size_t VolumeDim, typename Frame, typename DataType>
+void spatial_ricci_tensor(
+    const gsl::not_null<tnsr::ii<DataType, VolumeDim, Frame>*> ricci,
+    const tnsr::iaa<DataType, VolumeDim, Frame>& phi,
+    const tnsr::ijaa<DataType, VolumeDim, Frame>& deriv_phi,
+    const tnsr::II<DataType, VolumeDim, Frame>&
+        inverse_spatial_metric) noexcept {
+  if (UNLIKELY(get_size(get<0, 0>(*ricci)) != get_size(get<0, 0, 0>(phi)))) {
```

Comment:
does `destructive_resize_components(ricci, get_size(get<0, 0, 0>(phi));` work here?

Reply:
yup, changed to use that. thanks for pointing this out.